docs/2_BEST_PRACTICES.md

```diff
@@ -0,0 +1,161 @@
+
+You should try to keep your application state as flat as possible.
+
+**Do**
+
+```javascript
+const todosReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'ADD_TODO':
+      return [...state, action.payload]
+  }
+}
+```
+
+**Do NOT do**
+
+```javascript
+const todoReducer = (state = {todos: []}, action) => {
+  switch (action.type) {
+    case 'ADD_TODO':
+      return {
+        ...state,
+        todos: [...state.todos, action.payload],
+      }
+  }
+}
+```
+
+**Protip:** You can use the [handleActions](https://github.com/acdlite/redux-actions#handleactionsreducermap-defaultstate) method from [Redux Actions](https://github.com/acdlite/redux-actions) to create
+your reducers.
+
+### Action Creators
+
+Use [FSAs]((https://github.com/acdlite/flux-standard-action)) (short for Flux
+Standard Actions) to define more human-friendly, useful and simple actions.
+Another benefit from using FSAs is that you can deal with errors more
+easily because they are treated as a first class concept.
+
+**Do**
+
+```javascript
+const addTodo = (text) => ({
+  type: 'ADD_TODO',
+  text,
+})
+```
+
+**Do NOT do**
+
+```javascript
+const addTodo = (text) => ({
+  type: 'ADD_TODO',
+  payload: text,
+})
+```
+
+**Protip:** You can use the [createAction](https://github.com/acdlite/redux-actions#createactiontype-payloadcreator--identity-metacreator) method from [Redux Actions](https://github.com/acdlite/redux-actions)
+to create your actions creators.
+
+### Side-effects
+
+Use [Redux Saga](https://redux-saga.github.io/redux-saga/) and its [side-effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science))
+to separate your components from your application state in complex applications.
+This will allow you to simplify more complex applications, because your
+components can dispatch actions that will be handled by the side-effects.
+
+### Learn more
+
+* [Getting Started with Redux](https://egghead.io/courses/getting-started-with-redux)
+* [Building React Applications with Idiomatic Redux](https://egghead.io/courses/building-react-applications-with-idiomatic-redux)
+
+## Routing
+
+Use [React Router](https://github.com/ReactTraining/react-router) to define the
+routes for your application.
+
+**Protip:** You can use [React Router Redux](https://github.com/reactjs/react-router-redux)
+to connect React Router to Redux. It provides a reducer for the routing state
+and action creators for updating that state.
+
+**Note:** At the time of writing this [React Router v4](https://github.com/ReactTraining/react-router/tree/v4)
+is in beta. The major difference from v3 is that it embraces *declarative
+composability*. You should switch to v4 once it is stable.
+
+## Utilities
+
+Use [lodash](https://lodash.com/) or a similar utility library so that you do
+not need to write your own utility functions. Even though you might think that
+you do not need an utility library at first, you will need it sooner or later.
+
+## Packaging
+
+TODO
```

Comment:
Please consider adding a section that explains when you should put a package in `dependencies` and when it should be put in `devDependencies`.

Please also consider adding a section saying that would be nice if people try to actually build the production build from time to time. I'm currently fixing the latest breakage in one project, and that's only from a few days worth of changes.